packages/eslint-plugin-xcritical/src/rules/typescript.js

```diff
@@ -15,6 +15,29 @@ const typescriptRules = {
   '@typescript-eslint/no-type-alias': 'off',
   '@typescript-eslint/no-explicit-any': 'warn',
   '@typescript-eslint/quotes': ['error', 'single'],
-  '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true, allowTypedFunctionExpressions: true, allowHigherOrderFunctions: true }],
+  '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
+  '@typescript-eslint/no-unnecessary-condition': ['error', { ignoreRhs: true }],
+  '@typescript-eslint/no-inferrable-types': 0, // this is a temporary solution due to problems in TS (see: https://github.com/microsoft/TypeScript/pull/30593)
```

Comment:
Is it still necessary?

Reply:
Sounds like it's necessary. I mean this rule should be enabled